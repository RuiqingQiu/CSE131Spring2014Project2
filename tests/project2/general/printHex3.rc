function : void _printHex(int h, bool negative){
   if (h > 15){
      _printHex(h/16, negative);
      h = h % 16;
   }
   else{ if (negative){ h = h | 0xA; } }

   if (h < 10){ cout << h;}
   if (h == 10){ cout << "A";}
   if (h == 11){ cout << "B";}
   if (h == 12){ cout << "C";}
   if (h == 13){ cout << "D";}
   if (h == 14){ cout << "E";}
   if (h == 15){ cout << "F";}
}

function : void printHex(int h){
   cout << "0x";
   _printHex(h & 0x7FFFFFFF, h < 0);
   cout << endl;
}

function : int main(){
   printHex(0x12);
   printHex(0xABCDEF);
   printHex(012345);
   printHex(-1);
   printHex(-2);
   printHex(-0xF);
   return 0;
}
