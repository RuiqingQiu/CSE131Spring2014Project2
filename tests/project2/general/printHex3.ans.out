0x12
0xABCDEF
0x14E5
0xFFFFFFFF
0xFFFFFFFE
0xFFFFFFF1
